### DISCUSSION TITLE
Proper location to register custom elements

### DISCUSSION THREAD

**Discussion by apollo13** [unpoly/unpoly#660] [Q&A] [answered]
Hi there,

I am registering a very simple web component:
```
class OptionsMenu extends HTMLElement {
    constructor() {
      super()
      console.log("Constructed custom element", this)
    }
    
    connectedCallback() {
        console.log("Custom element added to page.");
      }
    
      disconnectedCallback() {
        console.log("Custom element removed from page.");
      }
    
      adoptedCallback() {
        console.log("Custom element moved to new page.");
      }    
}

window.customElements.define("options-menu", OptionsMenu)
```

and the logs show that it gets constructed twice for a single instance of `<options-menu>`. Any ideas what I am doing wrong?
![image](https://github.com/user-attachments/assets/5fcf442c-fdac-4f47-86f2-283989ae50ea)

**Comment by apollo13** [comment, marked as answer]
Haha, that was me beeing stupid, the element was indeed on the page twice. Sorry for the noise :)

That said: Would it make sense to register the elements in `up:framework:booted`?

**Reply by triskweline** [reply]
> Would it make sense to register the elements in up:framework:booted?

I believe custom elements do not care when they are defined. I personally would register them in my main script (when I'm also registering compilers), without waiting for an event.
